tests/unittests/Foundation/NSDictionaryTests.mm

```diff
@@ -160,6 +160,172 @@
     EXPECT_OBJCEQ(@"world", dict[@"hello"]);
 }
 
+TEST(NSDictionary, ReadWriteURL) {
+    NSURL* url = [NSURL fileURLWithPath:@"dictionaryTestData.txt" isDirectory:NO];
+    NSDictionary* dict = @{ @"key1" : @1, @"k2" : @"val", @"z" : @[ @5 ] };
+    EXPECT_TRUE([dict writeToURL:url atomically:NO]);
+    NSDictionary* read = [NSDictionary dictionaryWithContentsOfURL:url];
+    EXPECT_OBJCEQ(dict, read);
+}
+
+TEST(NSDictionary, Description) {
+    NSDictionary* empty = @{};
+    EXPECT_OBJCEQ(@"{\n}", [empty description]);
+
+    NSDictionary* dict = @{ @"A" : @[ @1, @{ @"Key" : @9 } ], @"Z" : @"Value", @"G" : @"ZValue" };
+    NSString* expected = @"{\n"
```

Comment:
fun fact!
the Dictionary description, when the dictionary contains no sets or other strange objects, should match the original OpenStep property list format!